### DISCUSSION TITLE
After webpack 5 migration not able to use local packages in a mono repo

### DISCUSSION THREAD

**Discussion by elangoRamar** [webpack/webpack#18211] [Q&A]
# Bug report
I have a mono repo (using lerna + yarn) with 2 packages, each having their own webpack configurations. And i am using packageA on the packageB.

```
repo
    /packages
       - packageA
             /src
             package.json
             webpack.config.json
       - packageB
             /src
             package.json
             webpack.config.json

```
And recently i have updated my webpack version,
from "webpack": "^4.41.0",  "webpack-cli": "^3.3.9" 
to "webpack": "^5.90.3", "webpack-cli": "^5.1.4",

I am able to install and build the projects separately without any issues. But when i try to import anything from packageA in my packageB, it's returning undefined error. And it was working fine in webpack 4.

I can't give link the repo here as it is very big. Am i missing any configuration on the migration process.

**Other relevant information:**
Node.js version: 20

**Comment by alexander-akait** [comment]
> But when i try to import anything from packageA in my packageB, it's returning undefined error.

Please provide it, also pleas run `npm ls webpack`/`yarn ls webpack` in the root of your project

**Comment by elangoRamar** [comment]
i am exporting initWithConfiguration method from packageA.
undefined error: `TypeError: Cannot read properties of undefined (reading 'initWithConfiguration')`

yarn list webpack:
```
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ packageA@1.0.4
│  ├─ webpack-encoding-plugin@0.3.1
│  │  └─ webpack@4.47.0
│  └─ webpack@5.90.3
├─ packageB@1.0.4
│  ├─ webpack-encoding-plugin@0.3.1
│  │  └─ webpack@4.47.0
│  └─ webpack@5.90.3
└─ @nestjs/cli@10.3.2
   └─ webpack@5.90.1
✨  Done in 0.58s.
```

**Reply by alexander-akait** [reply]
You have `webpack@4.47.0` and `webpack@5.90.3`, please ask `webpack-encoding-plugin` update deps to support webpack@5

**Reply by elangoRamar** [reply]
Okay
